central/reports/service/v2/singleton.go

```diff
@@ -12,8 +20,75 @@ var (
 )
 
 func initialize() {
-	metadataDS := metadataDataStore.Singleton()
-	svc = New(metadataDS)
+	metadataStore := metadataDS.Singleton()
+	scheduler := initializeScheduler(metadataStore)
+	svc = New(metadataStore, scheduler)
+}
+
+func initializeScheduler(metadataDataStore metadataDS.DataStore) schedulerV2.Scheduler {
+	ctx := sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS, storage.Access_READ_WRITE_ACCESS),
+			sac.ResourceScopeKeys(resources.WorkflowAdministration)))
+
+	scheduler := schedulerV2.Singleton()
+	reportConfigDatastore := reportConfigDS.Singleton()
+	queuePendingReports(ctx, scheduler, metadataDataStore, reportConfigDatastore)
```

Comment:
does this need to happen on the main thread? I.e. does central need to wait until every single pending or scheduled report is queued before it can take traffic?

Reply:
This doesn't need to happen on main thread. Moved to separate go routines.